dev/run-tests.py

```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/env python
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import re
+import sys
+import shutil
+import subprocess
+
+spark_proj_root = \
+    os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+user_home_dir = os.environ.get("HOME")
+
+sbt_maven_profile_args_env = "SBT_MAVEN_PROFILES_ARGS"
+amplab_jenkins_build_tool_env = "AMPLAB_JENKINS_BUILD_TOOL"
+amplab_jenkins_build_tool = os.environ.get(amplab_jenkins_build_tool_env)
+amplab_jenkins = os.environ.get("AMPLAB_JENKINS")
+
+resolving_re = "^.*[info].*Resolving"
+merging_re = "^.*[warn].*Merging"
+including_re = "^.*[info].*Including"
+sbt_output_filter = re.compile(resolving_re + "|" + 
+                               merging_re + "|" +
+                               including_re)
+
+
+def get_error_codes(err_code_file):
+    """Function to retrieve all block numbers from the `run-tests-codes.sh`
+    file to maintain backwards compatibility with the `run-tests-jenkins` 
+    script"""
+    
+    with open(err_code_file, 'r') as f:
+        err_codes = [e.split()[1].strip().split('=') 
+                     for e in f if e.startswith("readonly")]
+        return dict(err_codes)
+
+
+def rm_r(path):
+    """Given an arbitrary path properly remove it with the correct python
+    construct if it exists
+    - from: http://stackoverflow.com/a/9559881"""
+
+    if os.path.isdir(path):
+        shutil.rmtree(path)
+    elif os.path.exists(path):
+        os.remove(path)
+
+
+def lineno():
+    """Returns the current line number in our program
+    - from: http://stackoverflow.com/a/3056059"""
+
+    return inspect.currentframe().f_back.f_lineno
+
+
+def run_cmd(cmd):
+    """Given a command as a list of arguments will attempt to execute the
+    command and, on failure, print an error message"""
+
+    if not isinstance(cmd, list):
+        cmd = cmd.split()
+    try:
+        subprocess.check_output(cmd)
```

Comment:
Shouldn't this be `check_call()` and not `check_output()`? We don't care about capturing the output here, we just want it to be forwarded to stdout.